node/coroutines/src/client_task.rs

```diff
@@ -690,7 +697,9 @@ impl ClientTask {
         let (owner_uid, uid_to_authority_map) = self.client.get_uid_to_authority_map(index);
 
         if owner_uid.is_none() {
-            self.client.shard_client.pool.reset(None, None);
+            if let Some(pool) = &self.client.shard_client.pool {
```

Comment:
To my understanding only two scenarios are possible:
1) Validator: Pool and signer are not None, and consensus is running;
2) Non-validator; Pool and signer are None, and consensus is not running.
Therefore `process_payload_response` should never be called on a non-validating node, and if it does we should panic because that would be bug in our code.